### PR TITLE
Adds java-jdbc

### PR DESCRIPTION
**Pull request by jam01**
Adds byteman rules for intercepting calls to java.sql.Driver.connect and wraps the result in a TracingConnection from java-jdbc contrib project. Includes a simpler jdbc h2 test from that same project.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -50,6 +50,7 @@
     <version.io.opentracing.contrib.tracerresolver>0.1.5</version.io.opentracing.contrib.tracerresolver>
     <version.io.opentracing.contrib.web-servlet-filter>0.1.1</version.io.opentracing.contrib.web-servlet-filter>
     <version.io.opentracing.contrib.okhttp3>0.1.0</version.io.opentracing.contrib.okhttp3>
+    <version.io.opentracing.contrib.jdbc>0.0.9-SNAPSHOT</version.io.opentracing.contrib.jdbc>
```
Comment (objectiser) [inline pom.xml]: Should not be dependent upon snapshot version - is there a dependency on a pending PR?

File: rules/java-jdbc/pom.xml [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>${version.junit}</version>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>com.h2database</groupId>
+            <artifactId>h2</artifactId>
+            <version>1.4.197</version>
```
Comment (objectiser) [inline rules/java-jdbc/pom.xml]: Could you define the h2 version and dependency in the root pom.

File: rules/java-jdbc/src/test/java/io/opentracing/contrib/agent/jdbc/JdbcITest.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	public void test() throws Exception {
+		try (Scope ignored = getTracer().buildSpan("jdbc-test").startActive(true)) {
+
+			Connection connection = DriverManager.getConnection("jdbc:h2:mem:jdbc");
+			Statement statement = connection.createStatement();
+			statement.executeUpdate("CREATE TABLE employer (id INTEGER)");
+			connection.close();
+		}
+
+		List<MockSpan> spans = getTracer().finishedSpans();
```
Comment (objectiser) [inline rules/java-jdbc/src/test/java/io/opentracing/contrib/agent/jdbc/JdbcITest.java]: Could you add a comment regarding why 2 spans - just to make it clear one is for the parent span explicitly created, and the other for the db update. Alternatively could move the check into the try block after the `executeUpdate` and only check for the single span?

**Review by objectiser — COMMENTED** [review]
@malafeev Would you be able to give this a quick review - mainly to confirm that the rule is appropriate/sufficient to register tracing for jdbc?

File: pom.xml (hunk 1 above)
Comment (jam01) [reply pom.xml]: No pending PR, but that's the version on the master branch for that contrib project https://github.com/opentracing-contrib/java-jdbc/blob/master/pom.xml#L21

Advise if I should just use 0.0.8 which has been published and should be the same.

**Review by objectiser — APPROVED** [review]
If you could rebase to fix the conflicts. Will also wait a day or two to see if @malafeev has any comments.

**Review by malafeev — APPROVED** [review]

**Comment by malafeev** [comment]
Lgtm

**Comment by objectiser** [comment]
@jam01 If you could rebase to resolve the conflicts, I'll be able to merge. Thanks.

**Comment by objectiser** [comment]
@jam01 Thanks, and sorry for the delay.

**Comment by objectiser** [comment]
Will be included in 0.4.0 available soon.

**Pull request merged**